BDArmory/Control/MissileFire.cs

```diff
@@ -6264,7 +6264,8 @@ public void SendTargetDataToMissile(MissileBase ml, bool clearHeat = true)
                 if (clearHeat) heatTarget = TargetSignatureData.noTarget;
                 if (BDArmorySettings.DEBUG_MISSILES)
                     Debug.Log("[BDArmory.MissileData]: Sending targetInfo to heat Missile...");
-                ml.targetVessel = ml.heatTarget.vessel.gameObject.GetComponent<TargetInfo>();
+                var heatTgtVessel = ml.heatTarget.vessel.gameObject.GetComponent<TargetInfo>();
```

Comment:
This will still throw an NRE if the `gameObject` is `null`. I think what you want here is
```csharp
var heatTgtObject = ml.heatTarget.vessel.gameObject;
if (heatTgtObject) ml.targetVessel = heatTgtVessel.GetComponent<TargetInfo>();
```
assuming `ml.heatTarget.vessel` isn't `null`.
Similarly for the radar case below.